Missing spnego keytab for HDFS datanode

The systemd service `hadoop-hdfs-datanode` fail to start if the file `/etc/security/keytabs/spnego.service.keytab` is missing.

HDFS datanode log file content:

```
2022-02-24 15:27:02,615 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: registered UNIX signal handlers for [TERM, HUP, INT]
2022-02-24 15:27:02,881 INFO org.apache.hadoop.security.UserGroupInformation: Login successful for user dn/rpignolet-worker-01.novalocal@TDP.LOCAL using keytab file /etc/security/keytabs/dn.service.keytab
2022-02-24 15:27:03,044 INFO org.apache.hadoop.hdfs.server.datanode.checker.ThrottledAsyncChecker: Scheduling a check for [DISK]file:/var/lib/hdfs/dn
2022-02-24 15:27:03,160 WARN org.apache.hadoop.metrics2.impl.MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-datanode.properties,hadoop-metrics2.properties
2022-02-24 15:27:03,223 INFO org.apache.hadoop.metrics2.impl.MetricsSystemImpl: Scheduled Metric snapshot period at 10 second(s).
2022-02-24 15:27:03,223 INFO org.apache.hadoop.metrics2.impl.MetricsSystemImpl: DataNode metrics system started
2022-02-24 15:27:03,389 INFO org.apache.hadoop.hdfs.server.common.Util: dfs.datanode.fileio.profiling.sampling.percentage set to 0. Disabling file IO profiling
2022-02-24 15:27:03,391 INFO org.apache.hadoop.hdfs.server.datanode.BlockScanner: Initialized block scanner with targetBytesPerSec 1048576
2022-02-24 15:27:03,394 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: Configured hostname is rpignolet-worker-01.novalocal
2022-02-24 15:27:03,394 INFO org.apache.hadoop.hdfs.server.common.Util: dfs.datanode.fileio.profiling.sampling.percentage set to 0. Disabling file IO profiling
2022-02-24 15:27:03,407 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: Starting DataNode with maxLockedMemory = 0
2022-02-24 15:27:03,421 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: Opened streaming server at /0.0.0.0:9866
2022-02-24 15:27:03,422 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: Balancing bandwidth is 10485760 bytes/s
2022-02-24 15:27:03,422 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: Number threads for balancing is 50
2022-02-24 15:27:03,459 INFO org.eclipse.jetty.util.log: Logging initialized @1263ms
2022-02-24 15:27:03,517 INFO org.apache.hadoop.security.authentication.server.AuthenticationFilter: Unable to initialize FileSignerSecretProvider, falling back to use random secrets.
2022-02-24 15:27:03,519 INFO org.apache.hadoop.http.HttpRequestLog: Http request log for http.requests.datanode is not defined
2022-02-24 15:27:03,522 INFO org.apache.hadoop.http.HttpServer2: Added global filter 'safety' (class=org.apache.hadoop.http.HttpServer2$QuotingInputFilter)
2022-02-24 15:27:03,523 INFO org.apache.hadoop.http.HttpServer2: Added filter authentication (class=org.apache.hadoop.security.authentication.server.AuthenticationFilter) to context datanode
2022-02-24 15:27:03,524 INFO org.apache.hadoop.http.HttpServer2: Added filter authentication (class=org.apache.hadoop.security.authentication.server.AuthenticationFilter) to context logs
2022-02-24 15:27:03,524 INFO org.apache.hadoop.http.HttpServer2: Added filter authentication (class=org.apache.hadoop.security.authentication.server.AuthenticationFilter) to context static
2022-02-24 15:27:03,539 INFO org.apache.hadoop.http.HttpServer2: Jetty bound to port 40146
2022-02-24 15:27:03,540 INFO org.eclipse.jetty.server.Server: jetty-9.3.19.v20170502
2022-02-24 15:27:03,558 INFO org.apache.hadoop.security.authentication.server.AuthenticationFilter: Unable to initialize FileSignerSecretProvider, falling back to use random secrets.
2022-02-24 15:27:03,563 INFO org.eclipse.jetty.server.AbstractConnector: Started ServerConnector@584f54e6{HTTP/1.1,[http/1.1]}{localhost:40146}
2022-02-24 15:27:03,567 INFO org.apache.hadoop.hdfs.server.datanode.DataNode: Shutdown complete.
2022-02-24 15:27:03,567 ERROR org.apache.hadoop.hdfs.server.datanode.DataNode: Exception in secureMain
java.io.IOException: Problem starting http server
	at org.apache.hadoop.http.HttpServer2.start(HttpServer2.java:1165)
	at org.apache.hadoop.hdfs.server.datanode.web.DatanodeHttpServer.<init>(DatanodeHttpServer.java:141)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.startInfoServer(DataNode.java:954)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.startDataNode(DataNode.java:1404)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.<init>(DataNode.java:497)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.makeInstance(DataNode.java:2769)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.instantiateDataNode(DataNode.java:2677)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.createDataNode(DataNode.java:2719)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.secureMain(DataNode.java:2863)
	at org.apache.hadoop.hdfs.server.datanode.DataNode.main(DataNode.java:2887)
Caused by: javax.servlet.ServletException: javax.servlet.ServletException: Keytab does not exist: /etc/security/keytabs/spnego.service.keytab
	at org.apache.hadoop.security.authentication.server.KerberosAuthenticationHandler.init(KerberosAuthenticationHandler.java:179)
	at org.apache.hadoop.security.authentication.server.AuthenticationFilter.initializeAuthHandler(AuthenticationFilter.java:194)
	at org.apache.hadoop.security.authentication.server.AuthenticationFilter.init(AuthenticationFilter.java:180)
	at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:139)
	at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:873)
	at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:349)
	at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:778)
	at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:262)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:131)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:113)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.doStart(ContextHandlerCollection.java:161)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:131)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:113)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:131)
	at org.eclipse.jetty.server.Server.start(Server.java:422)
	at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:105)
	at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:61)
	at org.eclipse.jetty.server.Server.doStart(Server.java:389)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.apache.hadoop.http.HttpServer2.start(HttpServer2.java:1134)
	... 9 more
Caused by: javax.servlet.ServletException: Keytab does not exist: /etc/security/keytabs/spnego.service.keytab
	at org.apache.hadoop.security.authentication.server.KerberosAuthenticationHandler.init(KerberosAuthenticationHandler.java:139)
	... 33 more
2022-02-24 15:27:03,569 INFO org.apache.hadoop.util.ExitUtil: Exiting with status 1: java.io.IOException: Problem starting http server
```

I see it after the split between `hdfs` and `yarn`, before, the datanode was start after the YARN nodemanager is configured and the playbook which configure YARN nodemanager creates the spnego keytab that why it works before.

--- a/roles/hdfs/datanode/tasks/kerberos.yml
+++ b/roles/hdfs/datanode/tasks/kerberos.yml
@@ -12,3 +12,13 @@
     user: "{{ hdfs_user }}"
     group: "{{ hadoop_group }}"
     mode: "600"
+
+- import_role:
+    name: tosit.tdp.utils.kerberos
+    tasks_from: create_principal_keytab
+  vars:
+    principal: "HTTP/{{ ansible_fqdn }}"
+    keytab: "spnego.service.keytab"
+    user: "root"
+    group: "{{ hadoop_group }}"
+    mode: "640"

